source/features/extend-file-types-filter.tsx

```diff
@@ -0,0 +1,387 @@
+import {React} from 'dom-chef/react';
+import select from 'select-dom';
+import debounce from 'debounce-fn';
+import * as api from '../libs/api';
+import features from '../libs/features';
+import observeEl from '../libs/simplified-element-observer';
+import {getCleanPathname} from '../libs/utils';
+
+// --> Types and Interfaces
+interface PRFile {
+	fileName: string;
+	isDeleted: boolean;
+}
+
+interface PRFileTypes {
+	[fileType: string]: {
+		count: number;
+		deleted: number;
+	};
+}
+
+interface State {
+	prFiles: PRFile[];
+	prFileTypes: PRFileTypes;
+	selectedFileTypes: Set<string>;
+	shouldExtendFileType: boolean;
+}
+// <-- Types and Interfaces
+
+// --> State & Transitions
+const state: State = {
+	prFiles: [],
+	prFileTypes: {},
+	selectedFileTypes: new Set(),
+	shouldExtendFileType: true
+};
+
+const setState = (
+	diff: Partial<State>,
+	callback: (updated: boolean) => any,
+): void => {
+	const newState: State = {...state, ...diff};
+	if (JSON.stringify(state) === JSON.stringify(newState)) {
+		callback(false);
+		return;
+	}
+
+	Object.assign(state, newState);
+	callback(true);
+};
+
+const getIsShowingAllTypes = (): boolean => {
+	const numFileTypes = Object.keys(state.prFileTypes).length;
+	return state.selectedFileTypes.size !== numFileTypes;
+};
+
+const getIsShowingSomeTypes = (): boolean => {
+	return state.selectedFileTypes.size > 0;
+};
+// <-- State & Transitions
+
+// --> Helpers and Utils
+const sortObject = (unsorted: AnyObject): AnyObject => {
+	const sorted = {};
+	for (const key of Object.keys(unsorted).sort()) {
+		sorted[key] = unsorted[key];
+	}
+
+	return sorted;
+};
+
+const getPRFiles = async (): Promise<PRFile[]> => {
+	const pullUrl = getCleanPathname().replace('/pull/', '/pulls/');
+	const apiUrl = `repos/${pullUrl}?per_page=1000`;
+	const result = await api.v3(apiUrl);
+	return result.map(({status, filename}) => ({
+		fileName: filename,
+		isDeleted: status === 'removed'
+	}));
+};
+
+const getFileType = (
+	fileName: string,
+	shouldExtend: boolean = state.shouldExtendFileType,
+): string => {
+	const basename = fileName.split('/').pop();
+	const i = shouldExtend ? basename.indexOf('.') : basename.lastIndexOf('.');
+	return i < 0 ? null : basename.substr(i);
+};
+
+const groupPRFileTypes = (prFiles: PRFile[]): PRFileTypes => {
+	const grouped = {};
+	for (const {fileName, isDeleted} of prFiles) {
+		const fileType = getFileType(fileName);
+		if (!fileType) {
+			return grouped;
```

Comment:
```suggestion
			continue;
```

`return` will stop the loop, I'm not sure if that's your intention

Reply:
Thanks, left over from the previous function